### PR TITLE
adds react-flow package

### PR DESCRIPTION
**Pull request by JosephJNK**
Some teams are adopting Flow, but additional configuration is needed to get eslint to work with it.

### CODE REVIEW

File: packages/eslint-config-godaddy-flow/README.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+There are two ways to use this styleguide depending on your own tooling preference: directly using pre-included binaries or running `eslint` yourself with a custom `.eslintrc` config.
+
+##### 1. Use the pre-included binaries.
+
+These use _exactly_ the configuration defined in this  package (`eslint-config-godaddy-flow`) **with auto-fix enabled automatically.**
+
+``` js
+{
+  "scripts": {
+    "lint": eslint-godaddy-flow files/ you/ want-to/ lint/"
```
Comment (samshull) [inline packages/eslint-config-godaddy-flow/README.md]: Unclosed string in markdown

File: packages/eslint-config-godaddy-react-flow/README.md [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+There are two ways to use this styleguide depending on your own tooling preference: directly using pre-included binaries or running `eslint` yourself with a custom `.eslintrc` config.
+
+##### 1. Use the pre-included binaries.
+
+These use _exactly_ the configuration defined in this  package (`eslint-config-godaddy-react-flow`) **with auto-fix enabled automatically.**
+
+``` js
+{
+  "scripts": {
+    "lint": eslint-godaddy-react-flow files/ you/ want-to/ lint/"
```
Comment (samshull) [inline packages/eslint-config-godaddy-react-flow/README.md]: Unclosed string

**Comment by jcrugzz** [comment]
Give this a couple days to marinate. I know @indexzero wants a chance to review

File: packages/eslint-config-godaddy-react-flow/index.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
+module.exports = {
+  extends: [
+    require('./extends')('eslint-config-godaddy'),
```
Comment (indexzero) [inline packages/eslint-config-godaddy-react-flow/index.js]: Why wouldn't this be `eslint-config-godaddy-react`? 

``` js
require('./extends')('eslint-config-godaddy-react'),
```

File: packages/eslint-config-godaddy-react-flow/index.js [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    'react/jsx-pascal-case': 2,
+    'react/jsx-uses-react': 1,
+    'react/prefer-es6-class': 2,
+    //
+    // Whitespace rules for specific scenarios (e.g. JSX)
+    //
+    'react/jsx-curly-spacing': [2, 'always', {
+      spacing: { objectLiterals: 'never' }
+    }],
+    'jsx-quotes': [2, 'prefer-single'],
```
Comment (indexzero) [inline packages/eslint-config-godaddy-react-flow/index.js]: All of these rules should be inherited from `eslint-config-godaddy-react` no? If we add rules to that package I'd like to see them reflected here.

File: packages/eslint-config-godaddy-react-flow/index.js (hunk 3 above)
Comment (JosephJNK) [inline packages/eslint-config-godaddy-react-flow/index.js]: Makes sense, I'll make this change.

**Pull request merged**